pkg/reconciler/base.go

```diff
@@ -146,21 +149,31 @@ func NewGenericDeploymentReconciler[T kmeta.OwnerRefable, L Lister[T]](ctx conte
 	ownersLister ListerGetter[T, L],
 ) GenericDeploymentReconciler[T, L] {
 
-	informer := deploymentinformerv1.Get(ctx)
+	deplInformer := deploymentinformerv1.Get(ctx)
+	podInformer := podinformerv1.Get(ctx)
 
 	r := GenericDeploymentReconciler[T, L]{
 		SinkResolver:          resolver.NewURIResolverFromTracker(ctx, tracker),
 		Client:                k8sclient.Get(ctx).AppsV1().Deployments,
-		PodClient:             k8sclient.Get(ctx).CoreV1().Pods,
-		Lister:                informer.Lister().Deployments,
+		Lister:                deplInformer.Lister().Deployments,
+		PodLister:             podInformer.Lister().Pods,
 		GenericRBACReconciler: NewGenericRBACReconciler(ctx, ownersLister),
 	}
 
-	informer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+	deplInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
 		FilterFunc: controller.FilterControllerGVK(gvk),
 		Handler:    controller.HandleAll(adapterHandlerFn),
 	})
 
+	var outermostCtlrType T
+
+	podinformerv1.Get(ctx).Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: adapterPodWithAncestorOfKind(ctx, outermostCtlrType),
+		Handler: controller.HandleAll(
+			podOutermostAncestorHandlerFn(ctx, adapterHandlerFn),
+		),
+	})
```

Comment:
This is where we handle changes to Pod statuses (or any other change to Pod objects):

- Filter out Pods that do no have an ancestor of type `T` (where `T` is the type of the component being reconciled by this specific reconciler instance).
- If it's a match, pass the ancestor to `adapterHandlerFn` (enqueue it for reconciliation).